src/lib.rs

```diff
@@ -182,6 +172,21 @@ pub trait ToPublicKey: MiniscriptKey {
     /// Converts an object to a public key
     fn to_public_key(&self) -> bitcoin::PublicKey;
 
+    /// Convert an object to x-only pubkey
+    fn to_x_only_pubkey(&self) -> bitcoin::schnorr::PublicKey {
+        let pk = self.to_public_key();
+        bitcoin::schnorr::PublicKey::from(pk.key)
+    }
+
+    /// Serialize the key as bytes based on script context. Used when encoding miniscript into bitcoin script
+    fn push_to_builder<Ctx: ScriptContext>(&self, builder: script::Builder) -> script::Builder {
+        if Ctx::is_tapctx() {
+            builder.push_slice(&self.to_x_only_pubkey().serialize())
+        } else {
+            builder.push_slice(&self.to_public_key().to_bytes())
```

Comment:
This can be `builder.push_key(&self.to_public_key())` which avoids any allocations